akka-actor/src/main/scala/akka/actor/AbstractActor.scala

```diff
@@ -29,57 +86,138 @@ object AbstractActor {
  *   int count = 0;
  *
  *   public MyActor() {
- *     receive(ReceiveBuilder.
- *       match(Double.class, d -> {
+ *     receive(receiveBuilder()
+ *       .match(Double.class, d -> {
  *         sender().tell(d.isNaN() ? 0 : d, self());
- *       }).
- *       match(Integer.class, i -> {
+ *       })
+ *       .match(Integer.class, i -> {
  *         sender().tell(i * 10, self());
- *       }).
- *       match(String.class, s -> s.startsWith("foo"), s -> {
+ *       })
+ *       .match(String.class, s -> s.startsWith("foo"), s -> {
  *         sender().tell(s.toUpperCase(), self());
- *       }).build()
+ *       })
+ *       .build();
  *     );
  *   }
  * }
  * </pre>
  *
- * This is an EXPERIMENTAL feature and is subject to change until it has received more real world testing.
  */
-@ApiMayChange
 abstract class AbstractActor extends Actor {
 
-  private var _receive: Receive = null
+  /**
+   * Returns this AbstractActor's ActorContext
+   * The ActorContext is not thread safe so do not expose it outside of the
+   * AbstractActor.
+   */
+  def getContext(): AbstractActor.ActorContext = context.asInstanceOf[AbstractActor.ActorContext]
```

Comment:
Should we add a "Scala API" on context in Actor as well to help java devs not choosing that one? (might be applicable for more accessor-like methods on Actor as well)